### ISSUE TITLE
Add a 'bom' to support managing dependencies

### ISSUE THREAD

**Issue by mattrpav** [ops4j/org.ops4j.pax.web#1766]

**Comment by mattrpav** [comment]
It should include org.ops4j.pax.web groupId artifacts, and dependencies. Basically, anything listed in a pax-web-* feature, so downstream users can simply reference 'org.eclipse.jetty/jetty-server' and the version is managed.

For example:
jetty-continuation, jetty-deploy, jetty-http, jetty-io, jetty-plus, jetty-proxy, etc

**Comment by mattrpav** [comment]
Come to think of it, pax-web might need separate boms if there are any overlap of transitive deps for pax-web-jetty vs pax-web-tomcat vs pax-web-undertow.

**Comment by grgrzybek** [comment]
And the "dependencies" part is what I was afraid of... Because it'd include JavaEE APIs (even the fundamental one - `jakarta.servlet/jakarta.servlet-api` - but what if someone wants to use Geronimo/JBoss/SMX/Javax version of it?
I'll check and see where it's heading ;).

**Comment by mattrpav** [comment]
The pom section used in a bom is _dependencyManagement_. You can list all the servlet api options. 

The dependencyManagement does not mean _dependency_, it only sets version and scope when referenced by child poms in a dependency section. 

See: https://github.com/apache/karaf/blob/main/bom/pom.xml

**Comment by grgrzybek** [comment]
yes yes yes, I just work with BOMs for last few years and I know how messy they can be ;)
If Jetty/Tomcat/Undertow bundles _bring in_ their own deps, it's sometimes hard to choose.

I like the clean approach of https://repo1.maven.org/maven2/org/springframework/spring-framework-bom/5.3.23/spring-framework-bom-5.3.23.pom - it lists only _own_ managed dependencies.

**Comment by mattrpav** [comment]
@grgrzybek what is a scenario that breaks with boms?

If child project imports 2 boms that have different versions, maven will resolve one and the user has the option of using the resolved, or overriding version and scope.

The problem with only using ‘own’ managed deps is that there is now a gap from what is listed in the feature. For building offline repos (ie. pax-web + cxf) requires surgery.

**Comment by grgrzybek** [comment]
Yes, I understand. We could start with "project-bom" that lists only Pax Web managed deps and add optional, _convenient_ "all-bom" (or "dev-bom", or "user-bom") that lists current dependencies (for example if there was Jetty BOM, I could use it there.

The point is that I prefer managing my own dependencies than include one BOM. I trust Spring Framework BOM, but for example, Spring Boot BOM is just too much - I want to control more. But that's personal approach.

Let me thing about it - maybe you could create a PR with a suggestion?

**Comment by mattrpav** [comment]
Looping back to this, plan to start with simply a pax-web-bom-artifacts that lists artifacts created by pax-web, no samples, no tests, no dependencies.

**Comment by grgrzybek** [comment]
> Looping back to this, plan to start with simply a pax-web-bom-artifacts that lists artifacts created by pax-web, no samples, no tests, no dependencies.

I think that's a good start ;) Just list of pax web artifacts that could be used by developers.
